File: broker/src/main/scala/io/octopus/scala/factory/MQTTConnectionFactory.scala
package io.octopus.scala.factory

import io.handler.codec.mqtt.IMessage
import io.netty.channel.Channel
import io.octopus.broker.security.ReadWriteControl
import io.octopus.base.config.BrokerConfiguration
import io.octopus.base.interfaces.IAuthenticator
import io.octopus.base.queue.MsgQueue
import io.octopus.interception.BrokerNotifyInterceptor
import io.octopus.scala.broker.session.SessionResistor
import io.octopus.scala.broker.{MQTTConnection, PostOffice}

/**
 * connection create Factory
 * @param brokerConfig config
 * @param authenticator auth
 * @param sessionFactory sessionRegister
 * @param postOffice postOffice
 * @param interceptor interceptor
 */
class MQTTConnectionFactory(brokerConfig: BrokerConfiguration, authenticator: IAuthenticator, sessionFactory: SessionResistor,
                            postOffice: PostOffice, interceptor: BrokerNotifyInterceptor, readWriteControl:ReadWriteControl, msgQueue:MsgQueue[IMessage]){

  def create(channel: Channel): MQTTConnection = new MQTTConnection(channel,brokerConfig,authenticator,
    sessionFactory,postOffice,readWriteControl,msgQueue:MsgQueue[IMessage],interceptor)

}

File: broker/src/main/java/io/octopus/broker/MqttConnectionFactory.java
package io.octopus.broker;

import io.octopus.broker.config.BrokerConfiguration;
import io.octopus.broker.security.IAuthenticator;
import io.octopus.interception.BrokerNotifyInterceptor;
import io.netty.channel.Channel;

/**
 * Connection factory
 *
 * create connection for client
 */
public class MqttConnectionFactory {

    private final BrokerConfiguration brokerConfig;
    private final IAuthenticator authenticator;
    private final SessionRegistry sessionRegistry;
    private final PostOffice postOffice;
    private final BrokerNotifyInterceptor interceptor;

    MqttConnectionFactory(BrokerConfiguration brokerConfig, IAuthenticator authenticator,
                          SessionRegistry sessionRegistry, PostOffice postOffice, BrokerNotifyInterceptor interceptor) {
        this.brokerConfig = brokerConfig;
        this.authenticator = authenticator;
        this.sessionRegistry = sessionRegistry;
        this.postOffice = postOffice;
        this.interceptor = interceptor;
    }

    public MqttConnection create(Channel channel) {
        return new MqttConnection(channel, brokerConfig, authenticator, sessionRegistry, postOffice,interceptor);
    }
}
